dipy/viz/ui.py

```diff
@@ -0,0 +1,562 @@
+                msg = ("The ShowManager requires `CustomInteractorStyle` in"
+                       " order to use callbacks.")
+                raise TypeError(msg)
+
+            iren.add_callback(*callback, args=[self])
+
+    def add_callback(self, prop, event_type, callback, priority=0):
+        """ Adds a callback to a specific event for this UI component.
+        Parameters
+        ----------
+        prop : vtkProp
+        event_type : event code
+        callback : function
+        priority : int
+        """
+        # Actually since we need an interactor style we will add the callback
+        # only when this UI component is added to the renderer.
+        self._callbacks.append((prop, event_type, callback, priority))
+
+    def set_center(self, position):
+        """ Sets the center of the UI component
+        Parameters
+        ----------
+        position : (float, float)
+        """
+        msg = "Subclasses of UI must implement `set_center(self, position)`."
+        raise NotImplementedError(msg)
+
+    def set_visibility(self, visibility):
+        """ Sets visibility of this UI component. """
+        for actor in self.get_actors():
+            actor.SetVisibility(visibility)
+
+    def handle_events(self):
+        self.add_callback("LeftButtonPressEvent", self.left_button_click_callback)
+        self.add_callback("LeftButtonReleaseEvent", self.left_button_release_callback)
+        self.add_callback("RightButtonPressEvent", self.right_button_click_callback)
+        self.add_callback("RightButtonReleaseEvent", self.right_button_release_callback)
+        self.add_callback("MouseMoveEvent", self.mouse_move_callback)
+
+    @staticmethod
+    def left_button_click_callback(i_ren, obj, self):
+        self.left_button_state = "clicked"
+        i_ren.event.abort()
+
+    @staticmethod
+    def left_button_release_callback(i_ren, obj, self):
+        if self.left_button_state == "clicked":
+            self.on_left_mouse_button_pressed(i_ren, obj, self)
+        self.left_button_state = "released"
+
+    @staticmethod
+    def right_button_click_callback(i_ren, obj, self):
+        self.right_button_state = "clicked"
+        i_ren.event.abort()
+
+    @staticmethod
+    def right_button_release_callback(i_ren, obj, self):
+        if self.right_button_state == "clicked":
+            self.on_right_mouse_button_pressed(i_ren, obj, self)
+        self.right_button_state = "released"
+
+    @staticmethod
+    def mouse_move_callback(i_ren, obj, self):
+        if self.left_button_state == "clicked" or self.left_button_state == "dragging":
+            self.left_button_state = "dragging"
+            self.on_left_mouse_button_drag(i_ren, obj, self)
+        elif self.right_button_state == "clicked" or self.right_button_state == "dragging":
+            self.right_button_state = "dragging"
+            self.on_right_mouse_button_drag(i_ren, obj, self)
+        else:
+            self.on_mouse_hover(i_ren, obj, self)
```

Comment:
Unless you specifically add a prop to the list of `active_props` in the interactor (https://github.com/nipy/dipy/blob/master/dipy/viz/interactor.py#L76), the `MouseMove` callback are only called when one of the mouse buttons is held down (it actually works in your `viz_ui.py` example if you middle click the button, hold it and move the mouse). See https://github.com/nipy/dipy/blob/master/dipy/viz/interactor.py#L153